### ISSUE TITLE
When a chat window is not already open and user receives the message, the receiver sees  the message twice

### ISSUE THREAD

**Issue by vikasbunny** [rpaschoal/ng-chat#9]
When  a receiver gets a message , and the  conversation window is not already opened , the user will see the message twice.

This happens because the method  `onMessageReceived` in ng-chat.component.ts 

 ```
private onMessageReceived(user: User, message: Message)
    {
        if (user && message)
        {
            let chatWindow = this.openChatWindow(user);

            if (chatWindow){
                chatWindow.messages.push(message);

                this.scrollChatWindowToBottom(chatWindow);
            }
        }
    }
```

Here the method call  this.openChatWindow(user); loads the message from messageHistory i.e. `this.adapter.getMessageHistory`

And then again
   chatWindow.messages.push(message); appends the message.

The fix would be to keep a memory of whether the window returned from `this.openChatWindow(user);`
returned a new window or an already returned window. and depending on this flag , only push the message if it is an already opened window

**Comment by rpaschoal** [comment]
Hi @vikasbunny,

Thank you for raising this issue.

Are you using any kind of web sockets on your implementation (EG: socket.io, signalR)?

I am just wondering if this could be the case of notifying the user before persisting the message into the user history. This would avoid a double message issue.

Let me know your thoughts around this. I am just considering that other people might push the message first before persisting it to a history (And this was my initial implementation design too).

Cheers!

**Comment by rpaschoal** [comment]
@vikasbunny I think I see the problem, even if you notify first, there is a request to the history after you push that to the socket when opening the window. 👍

**Comment by rpaschoal** [comment]
@vikasbunny I have merged #10 but I will be doing a few changes on top of it before publishing the package as we need a setting to check if the history is being used or not (In cases where the history is not used the window would just be blank).

**Comment by vikasbunny** [comment]
@rpaschoal That will be great, thank you so much.

**Comment by rpaschoal** [comment]
@vikasbunny I've just published 0.1.0-beta.12 with the fix. Would you be able to check and see if that is working fine for you?

I've added a new "historyEnabled" setting but it defaults to true so you probably don't need to worry about it.
